### PR TITLE
remove dead read/write StarList c++ tickets/dm-9416

### PR DESCRIPTION
**Pull request by parejkoj**

### CODE REVIEW

File: include/lsst/jointcal/AstroUtils.h [hunk 1]
```diff
@@ -1,37 +1,15 @@
 // This may look like C code, but it is really -*- C++ -*-
-#ifndef USNOUTILS__H
-#define USNOUTILS__H
```
Comment (kfindeisen) [inline include/lsst/jointcal/AstroUtils.h]: This must be kept, but should be standardized to the [LSST convention](https://developer.lsst.io/coding/cpp_style_guide.html#header-files-must-include-a-construction-that-prevents-multiple-inclusion). Same with the `#endif` down below, obviously.

File: include/lsst/jointcal/AstroUtils.h [hunk 2]
```diff
@@ -1,37 +1,15 @@
 // This may look like C code, but it is really -*- C++ -*-
```
Comment (kfindeisen) [inline include/lsst/jointcal/AstroUtils.h]: Should be [parseable by emacs](https://developer.lsst.io/coding/cpp_style_guide.html#a-the-head-of-each-h-and-each-cc-file-must-begin-with-a-fixed-format-comment-declaring-the-file-contents-and-format-to-emacs).

File: include/lsst/jointcal/BaseStar.h [hunk 3]
```diff
@@ -72,28 +65,16 @@ double flux;
...
 
-
-
 int DecodeFormat(const char *FormatLine, const char *StarName);
 
 
-/* what concerns the BaseStarList's : */
-
-
+/* BaseStarList */
```
Comment (kfindeisen) [inline include/lsst/jointcal/BaseStar.h]: Redundant and uninformative comment; remove.

File: include/lsst/jointcal/StarList.cc [hunk 4]
```diff
@@ -18,138 +18,6 @@ namespace lsst {
 namespace jointcal {
```
Comment (kfindeisen) [inline include/lsst/jointcal/StarList.cc]: Nobody expects to include a `.cc` file; consider either renaming (if it must be used as a header file, which I assume is the case) or moving it to `src/`.

File: include/lsst/jointcal/AstroUtils.h (hunk 1 above)
Comment (parejkoj) [inline include/lsst/jointcal/AstroUtils.h]: Done, and filed DM-9495 to fix the rest of them.

File: include/lsst/jointcal/StarList.cc (hunk 4 above)
Comment (parejkoj) [inline include/lsst/jointcal/StarList.cc]: Oh dear! I didn't even notice that this file was in `include/`. Good catch!

**Pull request merged**